test/types/auth.test-d.ts

```diff
@@ -0,0 +1,130 @@
+/*!
+ * Copyright (c) 2015-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ *
+ */
+import { AuthTransaction, UserClaims, OktaAuth, TokenParams } from '@okta/okta-auth-js';
+import { expectType, expectAssignable } from 'tsd';
+
+const authClient = new OktaAuth();
+
+const authorizeOptions: TokenParams = {
+  sessionToken: '00p8RhRDCh_8NxIin-wtF5M6ofFtRhfKWGBAbd2WmE',
+  responseType: ['token', 'id_token'],
+  responseMode: 'okta_post_message',
+  redirectUri: 'https://some.com/redirect',
+  scopes: ['openid', 'email', 'profile'],
+  state: '8rFzn3MH5q',
+  nonce: '51GePTswrm',
+  idp: '0oa62b57p7c8PaGpU0h7',
+  idpScope: 'email,profile',
+  display: 'popup',
+  prompt: 'consent',
+  maxAge: 100,
+  pkce: false,
+  clientId: 'GHtf9iJdr60A9IYrR0jw',
+  ignoreSignature: false,
+  codeChallengeMethod: 'SHA-256',
+  codeVerifier: 'dBjftJeZ4CVP-mB92K27uhbUJU1p1r_wW1gFWFOEjXk',
+  authorizationCode: 'ff34f',
+  codeChallenge: 'E9Melhoa2OwvFrEMTJguCHaoeK1t8URWbuGJSstw-cM',
+  grantType: 'authorization_code',
+  interactionCode: 'r34r5',
+  loginHint: 'aa',
+  timeout: 10,
+  popupTitle: 'Hello Okta',
+};
+
+const authorizeOptions2: TokenParams = {
+  responseType: 'token',
+  idpScope: ['email', 'profile'],
+  maxAge: '100',
+};
+
+(async () => {
```

Comment:
So we need to wrap the test body in IIFE to test async APIs?

Reply:
Without wrapping linter returns error: `Top-level 'await' expressions are only allowed when the 'module' option is set to 'esnext' or 'system', and the 'target' option is set to 'es2017' or higher.ts(1378)`